scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py

```diff
@@ -22,184 +22,321 @@
+                fp.write(swap_entry)
+            sys.stdout.write(f"Swap entry added: {swap_entry}\n")
+    except:
+        raise MotrError(errno.EINVAL, f"Cant append f{FSTAB}\n")
+    finally:
+        swap_on(self)
+
+def del_swap_fstab_by_vg_name(self, vg_name):
+
+    swap_off(self)
+
+    cmd = f"sed -i '/{vg_name}/d' {FSTAB}"
+    execute_command(self, cmd)
+
+    swap_on(self)
+
+def create_swap(self, swap_dev):
+
+    sys.stdout.write(f"Make swap of {swap_dev}\n")
+    cmd = f"mkswap -f {swap_dev}"
+    execute_command(self, cmd)
+
+    sys.stdout.write(f"Test {swap_dev} swap device\n")
+    cmd = f"test -e {swap_dev}"
+    execute_command(self, cmd)
+
+    sys.stdout.write(f"Adding {swap_dev} swap device to {FSTAB}\n")
+    add_swap_fstab(self, swap_dev)
+
+
+def create_lvm(self, index, metadata_dev):
+    '''
+        1. validate /etc/fstab
+        2. validate metadata device file
+        3. check requested volume group exist
+        4. if exist, remove volume group and swap related with it.
+           because if user request same volume group with different device.
+        5. If not exist, create volume group and lvm
+        6. create swap from lvm
+    '''
 
-def create_lvm(node_name, index, metadata_dev):
     index = index + 1
+    node_name = self._server_id
     vg_name = f"vg_{node_name}_md{index}"
     lv_swap_name = f"lv_main_swap{index}"
     lv_md_name = f"lv_raw_md{index}"
+    swap_dev = f"/dev/{vg_name}/{lv_swap_name}"
+
+    sys.stdout.write(f"metadata device: {metadata_dev}\n")
+
+    sys.stdout.write(f"Checking for {FSTAB}\n")
+    validate_file(FSTAB)
+
+    sys.stdout.write(f"Checking for {metadata_dev}\n")
+    validate_file(metadata_dev)
+
+    cmd = f"fdisk -l {metadata_dev}"
+    execute_command(self, cmd)
+
     try:
-        validate_file(metadata_dev)
+        cmd = f"vgs {vg_name}"
+        execute_command(self, cmd)
+    except MotrError:
+        pass
+    else:
+        sys.stdout.write(f"Removing {vg_name} volume group\n")
 
-        cmd = f"fdisk -l {metadata_dev}"
-        execute_command(cmd, TIMEOUT_SECS)
+        del_swap_fstab_by_vg_name(self, vg_name)
 
-        cmd = f"wipefs --all --force {metadata_dev}"
-        execute_command(cmd, TIMEOUT_SECS)
+        cmd = f"vgchange -an {vg_name}"
+        execute_command(self, cmd)
 
-        cmd = f"pvcreate {metadata_dev}"
-        execute_command(cmd, TIMEOUT_SECS)
+        cmd = f"vgremove {vg_name} -ff"
+        execute_command(self, cmd)
 
-        cmd = f"vgcreate {vg_name} {metadata_dev}"
-        execute_command(cmd, TIMEOUT_SECS)
+    sys.stdout.write(f"Creating physical volume from {metadata_dev}\n")
+    cmd = f"pvcreate {metadata_dev} --yes"
+    execute_command(self, cmd)
 
-        cmd = f"vgchange --addtag {node_name} {vg_name}"
-        execute_command(cmd, TIMEOUT_SECS)
+    sys.stdout.write(f"Creating {vg_name} volume group from {metadata_dev}\n")
+    cmd = f"vgcreate {vg_name} {metadata_dev}"
+    execute_command(self, cmd)
```

Comment:
Why are these commands taken out of "try ... except ..." block?

Reply:
exception are raised intentionally to break execution if something fails. All, exception are handled in main function of motr_setup. which reports final error and call-trace of failure.
Only few exception are handled in motr_mini_prov.py which needs to handle.
Example:
```
  try:
        cmd = f"vgs {vg_name}"
        execute_command(self, cmd)
    except MotrError:
        pass
    else:
        sys.stdout.write(f"Removing {vg_name} volume group\n")

        del_swap_fstab_by_vg_name(self, vg_name)

        cmd = f"vgchange -an {vg_name}"
        execute_command(self, cmd)

        cmd = f"vgremove {vg_name} -ff"
        execute_command(self, cmd)
```